plugins/optimization-detective/storage/rest-api.php

```diff
@@ -169,28 +172,20 @@ function od_handle_rest_request( WP_REST_Request $request ) {
 	$post_id = $result;
 
 	/**
-	 * Fires whenever a URL Metric was successfully collected.
+	 * Fires whenever a URL Metric was successfully stored.
 	 *
-	 * @since 0.6.0
+	 * @since n.e.x.t
 	 *
-	 * @param array $context {
-	 *     Context about the successful URL Metric collection.
-	 *
-	 *     @var int                                   $post_id
-	 *     @var WP_REST_Request<array<string, mixed>> $request
-	 *     @var OD_Strict_URL_Metric                  $url_metric
-	 *     @var OD_URL_Metrics_Group                  $group
-	 *     @var OD_URL_Metrics_Group_Collection       $group_collection
-	 * }
+	 * @param OD_URL_Metric_Stored_Context $context Context.
 	 */
 	do_action(
-		'od_url_metric_collected',
-		array(
-			'post_id'                      => $post_id,
-			'request'                      => $request,
-			'url_metric'                   => $url_metric,
-			'url_metrics_group'            => $group,
-			'url_metrics_group_collection' => $group_collection,
+		'od_url_metric_stored',
+		new OD_URL_Metric_Stored_Context(
```

Comment:
Since the only reason for this class to exist is the filter, I think we should clarify that in the name, which could be confusing if you look at the class alone.

Either we go with `OD_URL_Metric_Context`, which sounds like a good name, but could have the risk of sounding more generic than it actually is. Or we go with `OD_URL_Metric_Stored_Filter_Context`, which keeps the scope narrow as it is now, but clarifies that it's for a filter. I think the latter was unclear to me when I first saw the class, and I was wondering what "stored context" is.

Reply:
How about eaed94f82f0933ac974f50acbeaa93b92abe9eab?

Reply:
WFM 👍